core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/parser/antlr/DataRepoInputExpressionValidationVisitor.scala

```diff
@@ -44,20 +50,55 @@ class DataRepoInputExpressionValidationVisitor(rootEntityType: Option[String],
     if (tableColumns.exists(_.getName == attributeName)) {
       Success()
     } else {
-      Failure(new RawlsException(s"Missing attribute `${attributeName}` on table `${tableModel.getName}`"))
+      Failure(new RawlsExceptionWithErrorReport(
+        ErrorReport(StatusCodes.BadRequest, s"Column `${attributeName}` does not exist on table `${tableModel.getName}`")
+      ))
     }
   }
 
-  // TODO: CA-939 support relationships once that information is published by data repo
   private def traverseRelationsAndGetFinalTable(currentTableModel: TableModel, relations: List[RelationContext]): Try[TableModel] = {
     relations match {
       case Nil => Success(currentTableModel)
-      case _ => Failure(new RawlsException(s"Relationships are not currently supported."))
+      case nextRelationContext :: remainingRelations => {
+        val nextRelationName = nextRelationContext.attributeName.name.getText
+        maybeGetNextTableFromRelation(currentTableModel, nextRelationName).flatMap(traverseRelationsAndGetFinalTable(_, remainingRelations))
```

Comment:
yeah, it did have the tail recursion (that you added?) there before, but with the flatmap it gave an error:

```
Error:(62, 5) could not optimize @tailrec annotated method traverseRelationsAndGetFinalTable: it contains a recursive call not in tail position
    relations match {
```

Doug and I removed it since we figured (as you point out) that most expressions should be pretty short.